src/object/cli_ec.c

```diff
@@ -289,11 +291,14 @@ obj_ec_recov_tgt_recx_nrs(struct obj_reasb_req *reasb_req,
 			obj_ec_parity_tgt_nr(oca), DP_RC(rc));
 		goto out;
 	}
-	for (tgt = 0, tgt_nr = 0; tgt < obj_ec_tgt_nr(oca); tgt++) {
-		if (obj_ec_tgt_in_err(fail_info->efi_tgt_list,
-				      fail_info->efi_ntgts, tgt))
+
+	for (i = 0, tgt = obj_ec_shard_idx(dkey_hash, oca, 0);
+	     i < obj_ec_tgt_nr(oca); i++, tgt = (tgt + 1) % obj_ec_tgt_nr(oca)) {
+		if (obj_ec_tgt_in_err(fail_info->efi_tgt_list, fail_info->efi_ntgts, tgt)) {
+			D_DEBUG(DB_TRACE, "tgt %ui not available\n", tgt);
```

Comment:
I'm wondering why you change it every where (logic shard -> physical shard), why don't keep the ec reasemble code same as before that only for logical shard, and we change to physical shard at one place?

if change it everywhere, need to check it very carefully if some places is missed.

Reply:
hmm, either we changed this or we fix the RPC format (use logical offset) and  fail_tgt (to use logical offset).

TBH, I  thought they are about the similar changes, but we do not need change the format by this way.

Though I will check further to see if either way are better.  Thanks.